CONTRIBUTING.md

```diff
@@ -108,6 +108,85 @@ Example:
   void sequenceNumber(long sequenceNumber);
 ```
 
+## Adding new functionality without breaking APIs
+Ideally, we'd want to add new functionality without breaking existing APIs, especially within the scope of the API modules that are being checked by [Revapi](https://revapi.org/).
+
+Let's assume we'd want to add a `createBranch(String name)` method to the `ManageSnapshots` API.
+
+The most straight-forward way would be to add the below code:
+
+```java
+public interface ManageSnapshots extends PendingUpdate<Snapshot> {
+  // existing code...
+
+  // introduces an API-breaking change
+  ManageSnapshots createBranch(String name);
+}
+```
+
+And then add the implementation:
+```java
+public class SnapshotManager implements ManageSnapshots {
+  // existing code...
+
+  @Override
+  public ManageSnapshots createBranch(String name, long snapshotId) {
+    updateSnapshotReferencesOperation().createBranch(name, snapshotId);
+    return this;
+  }
+}
+```
+
+### Checking for API breakages
+Running `./gradlew revapi` will flag this as an API-breaking change:
+```
+./gradlew revapi
+> Task :iceberg-api:revapi FAILED
+> Task :iceberg-api:showDeprecationRulesOnRevApiFailure FAILED
+
+1: Task failed with an exception.
+-----------
+* What went wrong:
+Execution failed for task ':iceberg-api:revapi'.
+> There were Java public API/ABI breaks reported by revapi:
+
+  java.method.addedToInterface: Method was added to an interface.
+
+  old: <none>
+  new: method org.apache.iceberg.ManageSnapshots org.apache.iceberg.ManageSnapshots::createBranch(java.lang.String)
+
+  SOURCE: BREAKING, BINARY: NON_BREAKING, SEMANTIC: POTENTIALLY_BREAKING
+
+  From old archive: <none>
+  From new archive: iceberg-api-1.4.0-SNAPSHOT.jar
+
+  If this is an acceptable break that will not harm your users, you can ignore it in future runs like so for:
+
+    * Just this break:
```

Comment:
newbie question: If we introduce a new API, the CI shall flag it. How does it get handled during automated build?

Reply:
it is handled the same way as on CI. `revapi` is being executed as part of the normal build, which then flags API/ABI-breaking changes